xarray/core/formatting_html.py

```diff
@@ -348,26 +359,31 @@ def dataset_repr(ds) -> str:
 
 
 def summarize_datatree_children(children: Mapping[str, DataTree]) -> str:
-    N_CHILDREN = len(children) - 1
-
-    # Get result from datatree_node_repr and wrap it
-    lines_callback = lambda n, c, end: _wrap_datatree_repr(
-        datatree_node_repr(n, c), end=end
-    )
-
-    children_html = "".join(
-        (
-            lines_callback(n, c, end=False)  # Long lines
-            if i < N_CHILDREN
-            else lines_callback(n, c, end=True)
-        )  # Short lines
-        for i, (n, c) in enumerate(children.items())
-    )
+    MAX_CHILDREN = OPTIONS["display_max_children"]
+    n_children = len(children)
+
+    children_html = []
+    for i, (n, c) in enumerate(children.items()):
+        if (
+            MAX_CHILDREN is None
```

Comment:
Isn't `MAX_CHILDREN` always an `int`? It's defined like that in `OPTIONS`.

Reply:
Yeah I was thinking originally that it should be nullable but I didn't see any other nullable options so then I was worried that that might be an antipattern. The case for null is to allow arbitrarily large trees, which obviously has performance implications that triggered this work in the first place. All that is to say: yes. It is always an int. I'll change the code to make that explicit.